### DISCUSSION TITLE
How to get the position and orientation in the world reference frame of the torso ?

### DISCUSSION THREAD

**Discussion by Tordjx** [upkie/upkie#353] [Software] [answered]
I would like to get the position and orientation in the world reference frame. I am aware that there are additional information in the info given by env.step() and env.reset(), but that's not what I seek. 
After some digging, I found the methods vulp::actuation::find_link_index() and vulp::actuation::get_position_link_in_world() in the vulp documentation, but I am unsure on how to proceed

**Comment by stephane-caron** [comment, marked as answer]
The values from these functions are reported by the Bullet interface (code [here](https://github.com/upkie/vulp/blob/5455596b58eb85a6739274dc1d4335fc81cc5b27/vulp/actuation/BulletInterface.cpp#L179)) in ``bullet/base/orientation`` (and ``.../position``) of the observation dictionary. (Of course these ground-truth values only appear in simulation.)

We can use [foxplot](https://pypi.org/project/foxplot/) to check out these values or explore the logs produced by the simulation or real-robot spine. We can also check it out live using [mpackview](https://pypi.org/project/mpackview/):

![image](https://github.com/upkie/upkie/assets/1189580/196757f8-5774-48dc-aa02-f92937ae992a)

The full observation dictionary is returned in ``info["spine_observation"]`` as done for instance in [this example](https://github.com/upkie/upkie/blob/ced785e74293d7090ea1247d6b221a008d4a265f/examples/wheeled_balancing_torque_control.py#L34).
